test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/jbpm/ProcessFailoverIntegrationTest.java

```diff
@@ -111,6 +108,79 @@ public void processFailoverTest() {
+        logger.debug("Wait for scale");
+        deploymentScenario.getKieServerDeployment().waitForScale();
+
+        logger.debug("Force delete (Kill) Kie server instance.");
+        deploymentScenario.getKieServerDeployment().deleteInstances(kieServerInstance);
+        logger.debug("Wait for scale");
+        deploymentScenario.getKieServerDeployment().waitForScale();
+
+        waitUntilProcessVariableIsChanged(longScriptPid);
+
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+        assertProcessVariable(longScriptPid, variableKey, variableValueTwo);
+        processServicesClient.signalProcessInstance(CONTAINER_ID, longScriptPid, SIGNAL_2_NAME, null);
+        assertProcessInstanceState(longScriptPid, org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED);
+
+        assertThat(deploymentScenario.getKieServerDeployment().getInstances()).doesNotContain(kieServerInstance);
+    }
+
+    private void waitUntilProcessVariableIsChanged(Long pid) {
+        LocalDateTime endTime = LocalDateTime.now().plusMinutes(3);
+
+        while (LocalDateTime.now().isBefore(endTime)) {
+            ProcessInstance pi = null;
+            try {
+                pi = processServicesClient.getProcessInstance(CONTAINER_ID, pid, true);
+            } catch (KieServicesException ex) {
+                //ok
+            }
+
+            if (pi != null && pi.getVariables().containsValue(variableValueTwo)) {
+                return;
+            }
+            try {
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException ex) {
+                logger.error("Thread sleep error.", ex);
+            }
+        }
+        throw new RuntimeException("Process variable is not changed.");
+    }
+
+    private void registerKieContainer() {
+        logger.debug("Register Kie Container to Kie Server");
+        KieServerInfo serverInfo = kieServicesClient.getServerInfo().getResult();
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, DEFINITION_PROJECT_NAME, DEFINITION_PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+    }
+
+    private Long startProcessInstance(String processInstance) {
+        logger.debug("Start process instance " + processInstance);
+        Long pid = processServicesClient.startProcess(CONTAINER_ID, processInstance, Collections.EMPTY_MAP);
+        assertThat(pid).isNotNull().isGreaterThan(0L);
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+        return pid;
+    }
+
     private void signalStartLongScript(Long pid) {
         new Thread(() -> {
             processServicesClient.signalProcessInstance(CONTAINER_ID, pid, SIGNAL_NAME, null);
```

Comment:
Here we should catch KieServicesHttpException and verify that error code is 5xx - so the redirection should take place.